libs/grok/src/test/java/org/elasticsearch/grok/GrokTests.java

```diff
@@ -439,4 +439,13 @@ public void testExponentialExpressions() {
         run.set(false);
         assertThat(e.getMessage(), equalTo("grok pattern matching was interrupted after [200] ms"));
     }
+
+    public void testUnicodeFieldnames() {
```

Comment:
I agree that more tests would be good. A common pattern we have in tests is to add a special assert method for something like this, then call that method from different tests cases. I would have tests for random alphanumeric, special characters (including each that are defined in the regex), and then something outside of the characters defined in the regex.

Reply:
@rjernst done I think :)

Note: 

* The `:` is in the regex but not a supported part of a field name because of the possibility of type definition in the Grok section making that impossible.
* Added a simple check for the unsupported `()` as an example of a char falling outside of the allowed range.